doc/userguide/performance/tuning-considerations.rst

```diff
@@ -24,26 +28,95 @@ The detection engine tries to split out separate signatures into
 groups so that a packet is only inspected against signatures that can
 actually match. As in large rule set this would result in way too many
 groups and memory usage similar groups are merged together. The
-profile setting controls how aggressive this merging is done. Higher
-is better but results in (much) higher memory usage.
+profile setting controls how aggressive this merging is done. The default setting of `high` usually is good enough.
 
 The "custom" setting allows modification of the group sizes:
 
 ::
 
     custom-values:
-      toclient-groups: 50
-      toserver-groups: 50
+      toclient-groups: 100
+      toserver-groups: 100
 
-In general, increasing will improve performance, but will lead to
-higher memory usage.
+In general, increasing will improve performance. It will lead to minimal increase in memory usage. 
+The default value for `toclient-groups` and `toclient-groups` with `detect.profile: high` is 75.
 
 detect.sgh-mpm-context: <auto|single|full>
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 The multi pattern matcher can have it's context per signature group
 (full) or globally (single). Auto selects between single and full
-based on the **mpm-algo** selected. ac and ac-bs use "single".
-All others "full". Setting this to "full" with AC requires a
-lot of memory: 32GB+ for a reasonable rule set.
+based on the **mpm-algo** selected. ac, ac-bs, ac-ks, hs default to "single". 
+Setting this to "full" with `mpm-algo: ac` or `mpm-algo: ac-ks` offers better performance. Setting this to "full" with `mpm-algo: hs` is not recommended as it leads to much higher startup time. Instead with Hyperscan you can use either `detect.profile: high` or bigger custom group size settings as explained above which offers better performance than `ac` and `ac-ks` even with `detect.sgh-mpm-context: full`.
+
+af-packet
+~~~~~~~~~
+
+If you are using `af-packet` (default on Linux) it is recommended that you use af-packet v3 for IDS/NSM deployments. For IPS it is recommended you stay with af-packet v2. To make sure af-packet v3 is used you can specifically enforce it in the `af-packet` config section of suricata.yaml like so:
+
+::
+
+ af-packet:
+  - interface: eth0
+    ....
+    ....
+    ....
+    # To use the ring feature of AF_PACKET, set 'use-mmap' to yes
+    use-mmap: yes
+    # Lock memory map to avoid it goes to swap. Be careful that over subscribing could lock
+    # your system
+    #mmap-locked: yes
+    # Use tpacket_v3 capture mode, only active if use-mmap is true
+    # Don't use it in IPS or TAP mode as it causes severe latency
+    tpacket-v3: yes
+
```

Comment:
maybe reference this in the other part above as I was wondering why they were not included :)